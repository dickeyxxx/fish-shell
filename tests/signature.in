function signature_test  --signature '
  signature_test <which> literal
  signature_test <which> [--counted...] [options] [<value>...]
  -x, --longX
  -y, --longY
  -oldOpt
'
  echo __{$which}__
  for name in (set -l)
    if [ $name != 'which' ]
      echo $name = $$name
    end
  end
end

signature_test 1
signature_test 2 -x
signature_test 3 --counted -x --counted
signature_test 4 --longX --counted value1 value2
signature_test 5 -oldOpt --longX -y --counted value3
signature_test 6 literal
