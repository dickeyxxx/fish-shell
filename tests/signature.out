__1__
argv = 1
__2__
argv = 2 -x
opt_longX = 1
__3__
argv = 3 --counted -x --counted
opt_counted = 2
opt_longX = 1
__4__
argv = 4 --longX --counted value1 value2
opt_counted = 1
opt_longX = 1
value = value1 value2
__5__
argv = 5 -oldOpt --longX -y --counted value3
opt_counted = 1
opt_longX = 1
opt_longY = 1
opt_oldOpt = 1
value = value3
__6__
argv = 6 literal
cmd_literal = 1
